instance/models/openedx_instance.py

```diff
@@ -274,9 +274,14 @@ def get_load_balancer_configuration(self):
         """
         if not self.active_appserver:
             return self.get_preliminary_page_config(self.ref.pk)
+        ip_address = self.active_appserver.server.public_ip
+        if not ip_address:
+            # The active appserver doesn't have a public IP address.  This means that the server
+            # has been terminated, so we don't show the preliminary page in this case, and simply
+            # deconfigure the backend instead.
+            return [], []
```

Comment:
@antoviaque You are right.  This _is_ and error.  I was assuming that this method would still be called for instances that have been turned down, but this isn't the case.  The clean-up code I added unsets the `load_balancing_server` field when an instance is turned down.  Moreover, the new `set_appserver_inactive()` method gets implicitly called when a VM is terminated, so the Appserver won't be active any more after being terminated.  This behaviour seems to make sense, and it means we should never get her.

I had a look at production, and the only instance this applies to is [this one](https://console.opencraft.com/instance/1860/edx-appserver/584/).  It is indeed in a weird state, since it is marked as running, but doesn't have an IP address.  It might be worthwhile to investigate how it got into that state.

I will change the code to explicitly log an error here.